framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -49,12 +50,154 @@ public static ContractsUtils getInstance(
     }
 
     /**
+     * Represents a pre- or postcondition that must be verified by
+     * {@code BaseTypeVisitor} or one of its subclasses. Automatically
+     * extracted from annotations with meta-annotations
+     * {@code @PreconditionAnnotation} or {@code @PostconditionAnnotation},
+     * such as {@code @RequiresNonNull} and {@code @EnsuresNonNull}.
+     * Can also be generated, such as in
+     * {@code LockVisitor.generatePreconditionsBasedOnGuards}.
+     */
+    public static class PreOrPostcondition {
+        /**
+         * The expression for which the condition must hold, such as
+         * {@code "foo"} in {@code @RequiresNonNull("foo")}.
+         */
+        public final String expression;
+        /**
+         * The name of the qualifier class that describes the condition that
+         * must hold.
+         */
+        public final String annotationString;
+
+        public PreOrPostcondition(String expression, String annotationString) {
+            this.expression = expression;
+            this.annotationString = annotationString;
+        }
+    }
+
+    /**
+     * An Iterable set of PreOrPostcondition entries.
+     * Implements the Iterable interface but not the Set interface.
```

Comment:
I'm not so sure about the usefulness of this class and this comment. Can you document more why this is better than just using `Set<PreOrPostcondition>`?